dulwich/objects.py

```diff
@@ -881,6 +881,23 @@ def sign(self, keyid: Optional[str] = None):
                     self.as_raw_string(), mode=gpg.constants.sig.mode.DETACH
                 )
 
+    def verify(self):
+        """Verify GPG signature for this tag (if it is signed).
+
+        Raises:
+          gpg.errors.BadSignatures: if GPG signature verification fails
+        """
+        if self._signature is None:
+            return
+
+        import gpg
+
+        with gpg.Context() as ctx:
+            unused_data, unused_result = ctx.verify(
```

Comment:
AFAICT this just verifies that the tag has a valid signature, but doesn't provide a way to verify that it's signed with the right key?

Reply:
By right key do you mean checking that the email on the key matches the commit author?

Reply:
I'll double check and add a test case, but if it's signed by a key which has not been imported into gpg (meaning its untrusted) then my understanding is that `ctx.verify` will fail due to `GPG_ERR_NO_PUBKEY`?

Reply:
The key being imported locally doesn't imply that it's trusted, or trusted for this particular purpose. Verifying that the signature was made by the expected key could either be done by allowing the caller to pass in key ids into ctx.verify() or by return the resultant key ids from Tag.verify().